### PR TITLE
minimal suite and test runner

### PR DESCRIPTION
**Pull request by lionelvillard**
- Add minimal suite and test runners
- TestContext includes injected client and informers
- Add the ability to invoke a subset of tests based on requirements levels (e.g --requirements-must=false)

### CODE REVIEW

File: pkg/framework/suite_runner.go [hunk 1]
```diff
@@ -59,5 +75,29 @@ func (s *suite) Run() {
 		s.Configure(&BaseConfig{})
 	}
 
+	cfg = s.enableInjection()
+
 	os.Exit(s.m.Run())
 }
+
+func (s *suite) enableInjection() *rest.Config {
```
Comment (n3wscott) [inline pkg/framework/suite_runner.go:83]: you can replace this with:

```
import 	"knative.dev/pkg/injection/sharedmain"

.. use:
	ctx := sharedmain.EnableInjectionOrDie(nil, nil)


```

Comment (lionelvillard) [reply pkg/framework/suite_runner.go:83]: I tried but I'm getting `flag redefined: kubeconfig` and I don't know where it's coming from.

**Pull request merged**